src/ucp/core/ucp_mm.inl

```diff
@@ -73,39 +73,62 @@ not_found:
 }
 
 static UCS_F_ALWAYS_INLINE void
-ucp_memh_put(ucp_context_h context, ucp_mem_h memh)
+ucp_memh_invalidate(ucp_context_h context, ucp_mem_h memh,
```

Comment:
1. I would revert to previous name (ucp_memh_put)
2. can we move some slow-path code to C file (e.g when no action is required, user memory handle, or rcache was disabled)

Reply:
1. it's not rename, ucp_memh_put is next function
2. we need to identify those cases before calling